flask_apispec/wrapper.py

```diff
@@ -37,21 +38,35 @@ def __call__(self, *args, **kwargs):
         return self.marshal_result(unpacked, status_code)
 
     def call_view(self, *args, **kwargs):
+        view_fn = self.func
         config = flask.current_app.config
         parser = config.get('APISPEC_WEBARGS_PARSER', flaskparser.parser)
+        # Delegate webargs.use_args annotations
         annotation = utils.resolve_annotations(self.func, 'args', self.instance)
         if annotation.apply is not False:
             for option in annotation.options:
-                schema = utils.resolve_schema(option['args'], request=flask.request)
-                parsed = parser.parse(schema, locations=option['kwargs']['locations'])
+                schema = utils.resolve_schema(option['argmap'], request=flask.request)
+                view_fn = parser.use_args(schema, **option['kwargs'])(view_fn)
+        # Delegate webargs.use_kwargs annotations
+        annotation = utils.resolve_annotations(self.func, 'kwargs', self.instance)
+        if annotation.apply is not False:
+            for option in annotation.options:
+                schema = utils.resolve_schema(option['argmap'], request=flask.request)
                 if getattr(schema, 'many', False):
```

Comment:
NOTE: What follows here is entirely to provide more informative error output, since this now shares the same failure cases as webargs which may break some existing code that uses flask-apispec.  My thought here was to be as helpful as possible in transition, but knowing that this adds nothing functional, I'm not sure if the value add is worth it.

At the judgement of the maintainer, I'd be okay doing one of 3 things:

1. Leaving this code as is
2. Rewriting to only run on failure (i.e. - `except` on the expected failures and just provide a better error message)
3. Remove this block entirely